cli/README.md

```diff
@@ -0,0 +1,154 @@
+# Constellation CLI design guide
+
+This document aims to outline the design principals behind the CLI code and packages.
+It should be consulted by developers when making changes to the code of packages in [`cli/internal/`](./internal/).
+
+## Basic layout
+
+[`cli/`](./) consists of 2 packages, [`cli/cmd/`](./cmd/), and [`cli/internal/`](./internal/), and a [`main.go`](./main.go), which serves as the entrypoint for the CLI.
+No new packages or files should be added to this base directory.
+
+[`cli/cmd/`](./cmd/) defines the root command of the CLI. Any changes concerning the root command, or additions like persistent flags (flags that can be accessed by all subcommands) should be added here.
+
+TODO: Discuss: Root command could be added to `cli/internal/cmd/`
```

Comment:
since root is not intended to be reused and is application code, it seems fine in its current place